src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,63 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.UserInvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MINIMUM_AGE = 18;
+    public static final int MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userNonNull(user);
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+        checkIfContainsInStorage(user);
+        return storageDao.add(user);
+    }
+
+    private void userNonNull(User user) {
```

Comment:
```suggestion
    private void checkUserNonNull(User user) {
```
method names should be verbs